docs/HowToFAQ/Errors-Resolver3.md

```diff
@@ -0,0 +1,678 @@
+_range_ expression. It serves to limit the values of the quantified variable(s)
+that are to be considered in evaluating the expression after the `::`. 
+As such, this range expression must always have a `bool` type.
+
+## **Error: arguments must have comparable types (got _type_ and _type_)**
+
+```dafny
+datatype D = D()
+const z := 0 == D()
+```
+
+All types have `==` and `!=` operations between two elements of the same type.
+And in cases of subtypes and bit-vector types there may be implicit conversions
+that allow members of two different but related types to be compared.
+But dissimilar types cannot be compared.
+
+## **Error: arguments to _op_ must have a common supertype (got _type_ and _type_)**
+
+<!-- TODO -- left and right operands of !! -->
+
+<!-- 2 instances -->
+
+## **Error: arguments must be of a set or multiset type (got _type_)**
+
+```dafny
+const z := 1 !! 2
+```
+
+The `!!` operator denotes disjointness of sets.
+It is only permitted between sets or multisets of the same element type.
+
+## **Error: arguments to rank comparison must be datatypes (got _type_ and _type_)**
+
+```dafny
+datatype D = D()
+class A {}
+method m(a: D, b: A) {
+  assert a < b;
+}
+```
+
+The `<` and `>` operators are used for traditional numeric comparison, 
+comparison of prefixes in sequences (just `<`),
+subset relations among sets,
+and for rank (structural depth) comparisons between values of the same datatype.
+When used for rank comparison, both operands must be values of the same datatype.
+
+<!--two instances, for < and for > -->
+
+## **Error: arguments to _expr_ must have a common supertype (got _type_ and _type_)**
+
+```dafny
+const x: ORDINAL
+const y: int
+const z := y < x 
+```
+
+For binary operators, the two operands must be able to be implicitly converted to the same supertype.
+For example, two different int-based subtypes would be converted to int, or two values of different
+classes that extend the same trait could be converted to values of that trait.
+Where Dafny cannot determine such a common supertype, the comparison is illegal and this error message results.
+
+<!--two instances, for < <= and for > >= -->
+
+## **Error: arguments to _op_ must be of a numeric type, bitvector type, ORDINAL, char, a sequence type, or a set-like type (instead got _type_ and _type_)**
+
+```dafny
+const x: map<int,int>
+const z := x < x 
+```
+
+The `<`, `<=`, `>=`, and `>` operators are used for traditional numeric comparison, 
+comparison of prefixes in sequences (just `<`),
+and subset relations among sets.
+But they are not used for comparing maps or reference values.
+
+<!--two instances, for < <= and for > >= -->
+
+## **Error: type of _op_ must be a bitvector type (instead got _type_)**
+
+```dafny
+const z :=  0 << 1
+```
+
+The `<<` and `>>` operators are left- and right-shift operations. 
+They shift a bit-vector value by a given integer number of bits.
```

Comment:
The number of bits can also have bit-vector type, as you mention below.

Reply:
I hope the distinction between an integer value and an int type and an int-base type are clear (and consistently used)